utils.go

```diff
@@ -0,0 +1,335 @@
+	for i, v := range collection {
+		cast[i] = mutator(v)
+	}
+	return cast
+}
+
+// SetReferencedEnumValue is a utility function that receives an enum getter , EnumFactory and applies a de-referenced result of the factory to a setter
+//
+// source is any function that receives a `EnumFactory` and returns an interface or error
+// parser is an EnumFactory
+// setter is a recipient of the function results
+func SetReferencedEnumValue[T interface{}](source func(parser serialization.EnumFactory) (interface{}, error), parser serialization.EnumFactory, setter func(t *T)) error {
+	val, err := source(parser)
+	if err != nil {
+		return err
+	}
+	if val != nil {
+		res := (val).(*T)
+		setter(res)
+	}
+	return nil
+}
+
+// SetCollectionOfReferencedEnumValue is a utility function that receives an enum collection source , EnumFactory and applies a de-referenced result of the factory to a setter
+//
+// source is any function that receives a `EnumFactory` and returns an interface or error
+// parser is an EnumFactory
+// setter is a recipient of the function results
+func SetCollectionOfReferencedEnumValue[T interface{}](source func(parser serialization.EnumFactory) ([]interface{}, error), parser serialization.EnumFactory, setter func(t []*T)) error {
+	val, err := source(parser)
+	if err != nil {
+		return err
+	}
+	if val != nil {
+		res := make([]*T, len(val))
+		for i, v := range val {
+			res[i] = (v).(*T)
+		}
+		setter(res)
+	}
+	return nil
+}
+
+// SetCollectionOfPrimitiveValue is a utility function that receives a collection of primitives , targetType and applies the result of the factory to a setter
+//
+// source is any function that receives a `EnumFactory` and returns an interface or error
+// targetType is a string representing the type of result
+// setter is a recipient of the function results
+func SetCollectionOfPrimitiveValue[T interface{}](source func(targetType string) ([]interface{}, error), targetType string, setter func(t []T)) error {
+	val, err := source(targetType)
+	if err != nil {
+		return err
+	}
+	if val != nil {
+		res := make([]T, len(val))
+		for i, v := range val {
+			res[i] = v.(T)
+		}
+		setter(res)
+	}
+	return nil
+}
+
+// SetCollectionOfReferencedPrimitiveValue is a utility function that receives a collection of primitives , targetType and applies the re-referenced result of the factory to a setter
+//
+// source is any function that receives a `EnumFactory` and returns an interface or error
+// parser is an EnumFactory
+// setter is a recipient of the function results
+func SetCollectionOfReferencedPrimitiveValue[T interface{}](source func(targetType string) ([]interface{}, error), targetType string, setter func(t []T)) error {
+	val, err := source(targetType)
+	if err != nil {
+		return err
+	}
+	if val != nil {
+		res := make([]T, len(val))
+		for i, v := range val {
+			res[i] = *(v.(*T))
+		}
+		setter(res)
+	}
+	return nil
+}
+
+// P converts a values to a pointer
+func P[T interface{}](t T) *T {
```

Comment:
should we make this private instead?

Reply:
@rkodev I think that last one was missed during your last changes.

Reply:
I'm divided on this one. The "heaviness" of creating pointers for scalars should be solved by a language feature or a dedicated type manipulation library. What I'm afraid of is if we make this public, people use it for code that has nothing to do with Kiota clients, and the days the language improves that scenario and we pull it, people complain for us pulling it. Thoughts?

Reply:
I think we can leave it out. I'll make the change on https://github.com/microsoft/kiota-abstractions-go/pull/35